src/app/data/page.tsx

```diff
@@ -117,6 +124,80 @@ function DataContent() {
     };
   }, [theme]);
 
+  useEffect(() => {
+    const updateFormattedAmount = async () => {
+      if (!transaction?.parsed) return;
+
+      const { recipients, validators } = transaction.parsed;
+
+      if (recipients && recipients[0]?.amount) {
+        if (transaction?.parsed?.mode === "transferToken") {
+          const result = await formatAssetAmount({
+            asset: {
+              chainId: selectedChain?.id || "",
+              isToken: true,
+              assetId: (transaction.raw as any).to,
+            },
+            amount: recipients[0].amount,
+            chainData: supportedChains,
+          });
+          setFormattedAmount(`${result.formatted} ${result.ticker}`);
+          return;
+        }
+
+        const result = await formatAssetAmount({
+          asset: {
+            chainId: selectedChain?.id || "",
+            isToken: false,
+          },
+          amount: recipients[0].amount,
+          chainData: supportedChains,
+        });
+        setFormattedAmount(`${result.formatted} ${result.ticker}`);
+        return;
+      }
+
+      if (validators?.target?.amount) {
+        const result = await formatAssetAmount({
+          asset: {
+            chainId: selectedChain?.id || "",
+            isToken: false,
+          },
+          amount: validators.target.amount,
+          chainData: supportedChains,
+        });
+        setFormattedAmount(`${result.formatted} ${result.ticker}`);
+        return;
+      }
+
+      setFormattedAmount("N/A");
+    };
+
+    updateFormattedAmount();
+  }, [transaction, selectedChain, supportedChains]);
+
+  useEffect(() => {
+    const updateFormattedFees = async () => {
+      if (!transaction?.parsed?.fees) return;
+
+      const result = await formatAssetAmount({
+        asset: {
+          chainId: selectedChain?.id || "",
+          isToken: false, // fees are always in native currency
```

Comment:
:warning: Not always true, at least for cosmos chains. We do support it, so sometimes fees could have a ticker different than the main chain's currency.

I don't know the full context and impact here, so maybe not important though.

Reply:
not really important at this stage, but I'll add a note for it